packages/ros2-test/CMakeLists.txt

```diff
@@ -65,5 +68,12 @@ target_compile_definitions(soss-ros2-test
   PRIVATE
     "ROS2__GEOMETRY_MSGS__TEST_CONFIG=\"${CMAKE_CURRENT_LIST_DIR}/resources/ros2__geometry_msgs.yaml\""
     "ROS2__TEST_MSGS__TEST_CONFIG=\"${CMAKE_CURRENT_LIST_DIR}/resources/${test_msgs_config_file}\""
+    "ROS2__TEST_DOMAIN__TEST_CONFIG=\"${CMAKE_CURRENT_LIST_DIR}/resources/ros2__domain_change.yaml\""
     "ROS2__ROSIDL__BUILD_DIR=\"${CMAKE_BINARY_DIR}/soss/rosidl/ros2/lib\""
 )
+
+# Windows dll dependencies installation
+if(WIN32)
+  find_file(MOCKDLL NAMES "soss-mock.dll" PATHS "${soss-mock_DIR}" PATH_SUFFIXES "lib" )
+  install (FILES ${MOCKDLL} DESTINATION "${CMAKE_INSTALL_PREFIX}/../bin")
```

Comment:
Just curious, is it not sufficient to source the workspace of the `soss-mock` build and the `soss-ros2-test` build at the same time? I guess copying this file isn't a big deal, but I would hope that it's avoidable.

Reply:
I think is not. Sourcing workspace doesn't set dll directory to system path enviroment variable. When Windows runs SOSS or any test it needs to load dll dependencies at the begining. It search in directories inside path in order to find that. Our aproach is to copy all dll dependencies (system handlers for example) that other executables will need inside a bin folder. Then we add that bin folder to the system path variable.

Reply:
Oh, I think I understand the problem now.

I believe the ideal way to solve this is by changing [this install command](https://github.com/osrf/soss/blob/93aaed95e7f8b5dc80da398969580a8033887134/packages/core/cmake/soss_install_middleware_plugin.cmake#L71-L76) to something like:

```
  install(
    TARGETS ${plugin_library_target}
    EXPORT  ${plugin_library_target}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    COMPONENT ${plugin_library_target}
  )
```

and then change [this line](https://github.com/osrf/soss/blob/93aaed95e7f8b5dc80da398969580a8033887134/packages/core/cmake/soss_install_middleware_plugin.cmake#L90) to be system-dependent. For UNIX it should remain the same, but for Windows it should be something like

```
    set(plugin_library_directory "../../../bin")
```

That way the libraries are installed in a way that's more conventional for Windows, and then the `.mix` files can accommodate that different convention.

Reply:
I think that is a better approach. Thanks @mxgrey. However, we have to be aware about system path max length.

Reply:
Oh right, I forgot about that limitation on Windows :weary: 

Taking that into account, I suppose there's some strong justification for copying the `.dll` (even if we move the `.dll`s into `bin`), as much as that hurts the pedant in me.

Reply:
So, then, are you ok with current changes?

Reply:
Sure, I think we can go ahead and copy the dll. It's only being done for the test packages anyway, which are optional to build.

Reply:
Circling back to this, I just remembered that colcon has a `--merge-install` flag that installs all the packages into the same install directory. I think that could be used to deal with this issue pretty nicely, without the Windows PATH limit being a problem.

Reply:
I think is a good solution. I've been testing with `--merge-install` option and everything is ok. I've updated the doc file with this and also removed local_setup_windows.bat script because is no longer necessary.